tests/astimezone_test.py

```diff
@@ -0,0 +1,247 @@
+            datetime_tz(2021, 1, 1, 9, 0, 0, tzinfo=ZoneInfo('UTC')),
+        ),
+        (
+            datetime_cet(2021, 1, 1, 10, 0, 0, tzinfo=ZoneInfo('CET')),
+            ZoneInfo('EST'),
+            datetime_tz(2021, 1, 1, 4, 0, 0, tzinfo=ZoneInfo('EST')),
+        ),
+        (
+            datetime_cet(2021, 1, 1, 10, 0, 0, tzinfo=ZoneInfo('CET')),
+            ZoneInfo('CET'),
+            datetime_tz(2021, 1, 1, 10, 0, 0, tzinfo=ZoneInfo('CET')),
+        ),
+        # Test astimezone on datetime_tz
+        (
+            datetime_tz(2021, 1, 1, 10, 0, 0, tzinfo=ZoneInfo('UTC')),
+            ZoneInfo('CET'),
+            datetime_tz(2021, 1, 1, 11, 0, 0, tzinfo=ZoneInfo('CET')),
+        ),
+        (
+            datetime_tz(2021, 1, 1, 10, 0, 0, tzinfo=ZoneInfo('CET')),
+            ZoneInfo('UTC'),
+            datetime_tz(2021, 1, 1, 9, 0, 0, tzinfo=ZoneInfo('UTC')),
+        ),
+    ]
+)
+def test_astimezone(
+    original_dt: datetime_tz, target_tz: ZoneInfo, expected_dt: datetime_tz
+) -> None:
+    """Test that astimezone returns the correct datetime with proper timezone conversion."""
+    result = original_dt.astimezone(target_tz)
+
+    # Check that the result is of the same class as the original
+    assert isinstance(result, datetime_tz)
+
+    # Check that the timezone conversion is correct
+    assert result.year == expected_dt.year
+    assert result.month == expected_dt.month
+    assert result.day == expected_dt.day
+    assert result.hour == expected_dt.hour
+    assert result.minute == expected_dt.minute
+    assert result.second == expected_dt.second
+    assert result.microsecond == expected_dt.microsecond
+    assert result.tzinfo == expected_dt.tzinfo
+
+
+def test_astimezone_converts_type() -> None:
+    """Test that astimezone converts to datetime_tz when changing timezones."""
+    utc_dt = datetime_utc(2021, 1, 1, 10, 0, 0, tzinfo=ZoneInfo('UTC'))
+    result_utc = utc_dt.astimezone(ZoneInfo('CET'))
+    assert isinstance(result_utc, datetime_tz)
+
+    cet_dt = datetime_cet(2021, 1, 1, 10, 0, 0, tzinfo=ZoneInfo('CET'))
+    result_cet = cet_dt.astimezone(ZoneInfo('UTC'))
+    assert isinstance(result_cet, datetime_tz)
+
+    local_dt = datetime_local.now()
+    result_local = local_dt.astimezone(ZoneInfo('CET'))
+    assert isinstance(result_local, datetime_tz)
+
+    tz_dt = datetime_tz(2021, 1, 1, 10, 0, 0, tzinfo=ZoneInfo('UTC'))
+    result_tz = tz_dt.astimezone(ZoneInfo('CET'))
+    assert isinstance(result_tz, datetime_tz)
+
+
+def test_deepcopy_astimezone() -> None:
+    """Test that astimezone works correctly with deepcopy."""
+    original = datetime_utc(2021, 1, 1, 10, 0, 0, tzinfo=ZoneInfo('UTC'))
+
+    # Deep copy the original
+    copied = deepcopy(original)
+
+    # Apply astimezone to the copy
+    result = copied.astimezone(ZoneInfo('CET'))
+
+    # Check that the result is correct
+    assert isinstance(result, datetime_tz)
+    assert result.hour == 11  # UTC+1 for CET in January
```

Comment:
`CET` is always `UTC +1`, even in the summer. In the summer we use `CEST` in central europe, which is `UTC +2`

```suggestion
    assert result.hour == 11
```